build_tools/python/benchmark_suites/iree/vulkan_nvidia_benchmarks.py

```diff
@@ -0,0 +1,86 @@
+## Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Defines IREE Vulkan NVIDIA benchmarks."""
+
+from typing import List, Tuple, Sequence
+from benchmark_suites.iree import module_execution_configs
+from e2e_test_framework import unique_ids
+from e2e_test_framework.definitions import common_definitions, iree_definitions
+from e2e_test_framework.device_specs import device_collections
+from e2e_test_framework.models import model_groups
+import benchmark_suites.iree.utils
+
+
+class Linux_Vulkan_NVIDIA_Benchmarks(object):
+  """Benchmarks on Linux Vulkan NVIDIA devices."""
+
+  AMPERE_TARGET = iree_definitions.CompileTarget(
+      target_architecture=common_definitions.DeviceArchitecture.NVIDIA_AMPERE,
+      target_backend=iree_definitions.TargetBackend.VULKAN_SPIRV,
+      target_abi=iree_definitions.TargetABI.VULKAN_LINUX)
+  PASCAL_TARGET = iree_definitions.CompileTarget(
+      target_architecture=common_definitions.DeviceArchitecture.NVIDIA_PASCAL,
+      target_backend=iree_definitions.TargetBackend.VULKAN_SPIRV,
+      target_abi=iree_definitions.TargetABI.VULKAN_LINUX)
+
+  COMPILE_EXTRA_FLAGS = [
+      "--iree-stream-resource-index-bits=64", "--iree-vm-target-index-bits=64",
+      "--iree-preprocessing-pass-pipeline=builtin.module(func.func("
```

Comment:
This is a lot of custom stuff to get this to compile. Generally we want `iree-compile` to just work. Why is all this necessary? Should these configurations be tagged with "experimental flags"?

@MaheshRavishankar since you've had opinions about this sort of thing in the past.

Reply:
I think there is a difference here. This configuration is not something we are _experimenting_ a new compiler feature or whatever, it's actually the way we compile for _production_ for stable diffusion via nod.ai distribution channels. I'd say it's a different level so worth having special considerations w.r.t. support and guarantees.  I agree with the general direction of making the compiler just works so we don't need various flags, but it's still afar to reach that goal. While in the meanwhile, in the past few weeks I have been fixing quite a few regressions regarding this already, including https://github.com/openxla/iree/issues/13051, https://github.com/openxla/iree/issues/12670, https://github.com/openxla/iree/issues/12634, etc. Having CI coverage would avoid such continuous breakages for downstream deployment and avoid always spending a long time trying to triage and fix bugs afterwards. :)

Reply:
Ping on this to see if there are further comments. I think I can add `production` in the tags if that makes things clearer.

Reply:
If we're not able to turn them on by default then they seems pretty experimental, regardless of where people are using them. I think "production" would communicate the wrong thing for the same reason. I would still prefer to tag these as "experimental flags", since that's consistent with all our other custom flag configs, but if you want to call them "custom flags" or something instead, that's fine.

Reply:
SG. Added "experimental-flags" label.